lib/web_ui/lib/src/engine/canvaskit/native_memory.dart

```diff
@@ -0,0 +1,218 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:js_interop';
+import 'package:meta/meta.dart';
+
+import '../../engine.dart' show Instrumentation;
+import '../util.dart';
+import 'canvaskit_api.dart';
+
+/// Collects native objects that weren't explicitly disposed of using
+/// [UniqueRef.dispose] or [CountedRef.unref].
+SkObjectFinalizationRegistry _finalizationRegistry = createSkObjectFinalizationRegistry(
```

Comment:
I mentioned this offline, but perhaps it would be a bit less fragile to do the following:
* Make this variable nullable
* Guard the construction of this object by `browserSupportsFinalizationRegistry`, otherwise return null
* Have any uses of this object (i.e. `register` below) do a null check instead of `browserSupportsFinalizationRegistry`
That feels a little less fragile, because right now if someone accidentally touches this variable without checking `browserSupportsFinalizationRegistry` first they could cause things to explode. A nullable variable would let the static type system save you from accidentally doing that.

Reply:
Happy to fix it as a follow-up. For now there's no risk of this variable being touched by accident, because it's private. But in general, I like the idea of explicit lazy initialization.